target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac

```diff
@@ -139,4 +139,16 @@ case "$board" in
 		[ "$PHYNBR" = "0" ] && macaddr_add $hw_mac_addr 1 > /sys${DEVPATH}/macaddress
 		[ "$PHYNBR" = "1" ] && macaddr_add $hw_mac_addr 2 > /sys${DEVPATH}/macaddress
 		;;
+	unielec,u7981-01*)
+		case "$(cmdline_get_var root)" in
+		/dev/mmc*)
+			addr=$(mmc_get_mac_binary factory 0x4)
+			;;
+		*)
+			addr=$(mtd_get_mac_binary factory 0x4)
```

Comment:
I don't understand if factory data on this board is stored as UBI volume or MTD partition for the SPI-NAND variant.
In either case, if both bands (2.4GHz, 5GHz) use the same MAC address, you don't need to set it from userspace, using NVMEM in DTS should be enough.

Reply:
Thanks for your tips.I'm making the changes as requested.

Reply:
Sorry I don't quite understand what you mean.
The meaning of this code is: When using the U7981-01-nand board, the mac address is stored in the mtd3 (factory) partition. When using the U7981-01-emmc board, the mac address is stored in the /dev/mmcblk0p3 (factory) partition. middle

> I don't understand if factory data on this board is stored as UBI volume or MTD partition for the SPI-NAND variant. In either case, if both bands (2.4GHz, 5GHz) use the same MAC address, you don't need to set it from userspace, using NVMEM in DTS should be enough.

Reply:
All requirements have been modified,please check.
Thanks.

Reply:
> the mac address is stored in the mtd2 (factory) partition

Ok. So is there also a UBI volume called `factory` at the same time for the WiFi EEPROM as you defined
```
			caldata_extract_ubi "factory" 0x0 0x1000
```
in [`target/linux/mediatek/filogic/base-files/etc/hotplug.d/firmware/11-mt76-caldata`](https://github.com/openwrt/openwrt/pull/14834/files#diff-38e9c363b260069ffdaf4e802b9d152578b11e9aa74117abc11055deac77e365)?